nucypher/blockchain/eth/deployers.py

```diff
@@ -505,21 +505,68 @@ class StakingEscrowDeployer(BaseContractDeployer, UpgradeableContractMixin, Owna
+    init_steps = ('stub_deployment', 'dispatcher_deployment')
+    preparation_steps = ('contract_deployment', 'dispatcher_retarget')
     activation_steps = ('approve_reward_transfer', 'initialize')
     deployment_steps = preparation_steps + activation_steps
     _proxy_deployer = DispatcherDeployer
 
-    def __init__(self, test_mode: bool = False, *args, **kwargs):
+    def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.__dispatcher_contract = None
 
         token_contract_name = NucypherTokenDeployer.contract_name
+        policy_manager_proxy_name = PolicyManagerDeployer._proxy_deployer.contract_name
+        policy_manager_contract_name = PolicyManagerDeployer.contract_name
+        adjudicator_proxy_name = AdjudicatorDeployer._proxy_deployer.contract_name
+        adjudicator_contract_name = AdjudicatorDeployer.contract_name
+        worklock_contract_name = WorklockDeployer.contract_name
         self.token_contract = self.blockchain.get_contract_by_name(registry=self.registry,
                                                                    contract_name=token_contract_name)
-        self.test_mode = test_mode
+        try:
+            self.policy_manager = self.blockchain.get_contract_by_name(registry=self.registry,
+                                                                       contract_name=policy_manager_contract_name,
+                                                                       proxy_name=policy_manager_proxy_name)
+        except self.registry.UnknownContract:
+            self.policy_manager = None
+
+        try:
+            self.adjudicator = self.blockchain.get_contract_by_name(registry=self.registry,
+                                                                    contract_name=adjudicator_contract_name,
+                                                                    proxy_name=adjudicator_proxy_name)
+        except self.registry.UnknownContract:
+            self.adjudicator = None
+
+        try:
+            self.worklock = self.blockchain.get_contract_by_name(registry=self.registry,
+                                                                 contract_name=worklock_contract_name)
+        except self.registry.UnknownContract:
+            self.worklock = None
+
+    def _deploy_stub(self, gas_limit: int = None, confirmations: int = 0, **overrides):
+        constructor_kwargs = {
+            "_hoursPerPeriod": self.economics.hours_per_period,
+            "_minLockedPeriods": self.economics.minimum_locked_periods,
+            "_minAllowableLockedTokens": self.economics.minimum_allowed_locked,
+            "_maxAllowableLockedTokens": self.economics.maximum_allowed_locked
+        }
+        constructor_kwargs.update(overrides)
+        constructor_kwargs = {k: v for k, v in constructor_kwargs.items() if v is not None}
+        # Force use of the token address from the registry
```

Comment:
Can you elaborate a bit on what context is being handled here?

Reply:
This is deployment of stub and it uses some economics variable. We have logic for normal deployment when we can override some of those values. So here is the same: we can override values or add or even remove them from constructor if needed